File: legacy/src/main/java/com/xiaomi/mipush/sdk/BridgeActivity.java
package com.xiaomi.mipush.sdk;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;
import androidx.core.os.BundleCompat;
import com.xiaomi.channel.commonutils.logger.LoggerInterface;
import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.push.service.PushConstants;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/BridgeActivity.class */
public class BridgeActivity extends Activity {
    @Override // android.app.Activity
    protected void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        Window window = getWindow();
        WindowManager.LayoutParams attributes = window.getAttributes();
        attributes.height = 1;
        attributes.width = 1;
        attributes.gravity = 51;
        window.setAttributes(attributes);
    }

    @Override // android.app.Activity
    protected void onResume() {
        Intent intent;
        super.onResume();
        try {
            try {
                Intent intent2 = getIntent();
                Bundle extras = intent2 != null ? intent2.getExtras() : null;
                if (extras != null && (intent = BundleCompat.getParcelable(extras, PushConstants.MIPUSH_EXTRA_INTENT_PAYLOAD, Intent.class)) != null) {
                    PushMessageHandler.addJob(getApplicationContext(), intent);
                }
            } catch (Exception e) {
                MyLog.e(e);
            }
        } finally {
            finish();
        }
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/mipush/sdk/BridgeActivity.kt
package com.xiaomi.mipush.sdk

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.core.os.BundleCompat
import com.xiaomi.channel.commonutils.logger.MyLog
import com.xiaomi.push.service.PushConstants

/*
 * Stock reference: com.xiaomi.xmsf 7.4.67-C (versionCode 70004067),
 * split-XiaomiServiceFrameworkCN-master.apk sha256 444e9f128591e04e38672bfe44a246ab3fa97ae68e95882839d8a7afe766df2b,
 * JADX path: com.xiaomi.xmsf/stock/split-XiaomiServiceFrameworkCN-master/sources/com/xiaomi/mipush/sdk/BridgeActivity.java
 * No current same-path BridgeActivity.java is present in the 2026-04-13 current override.
 */
class BridgeActivity : Activity() {
    override fun onCreate(bundle: Bundle?) {
        super.onCreate(bundle)
        val window = window
        val attributes = window.attributes
        attributes.height = 1
        attributes.width = 1
        attributes.gravity = 51
        window.attributes = attributes
    }

    override fun onResume() {
        super.onResume()
        try {
            try {
                val extras = intent?.extras
                val payloadIntent = if (extras != null) {
                    BundleCompat.getParcelable(extras, PushConstants.MIPUSH_EXTRA_INTENT_PAYLOAD, Intent::class.java)
                } else {
                    null
                }
                if (payloadIntent != null) {
                    PushMessageHandler.addJob(applicationContext, payloadIntent)
                }
            } catch (e: Exception) {
                MyLog.e(e)
            }
        } finally {
            finish()
        }
    }
}
